tests/zfs-tests/tests/functional/cli_root/zpool_reopen/zpool_reopen.shlib

```diff
@@ -0,0 +1,140 @@
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2016, 2017 by Intel Corporation. All rights reserved.
+# Copyright (c) 2017 Open-E, Inc. All Rights Reserved.
+#
+
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/tests/functional/cli_root/zpool_reopen/zpool_reopen.cfg
+
+# part of logic borrowed from functional/fault test
```

Comment:
All of the functionality added here looks like it would be useful to other test cases.  I'd suggest adding it instead to `libtest.shlib` and clearly documenting their functionality with a block comment above the functions.  Please try and co-locate similar functions where possible.

If you're feeling really ambitious I wouldn't be opposed to pulling all of the blkdev/scsi helper functions out in to their own `blkdev.shlib`.  We'd specifically like to write more test cases which verify correct behavior when there are block device failures so having this all in one place would be convenient.

Reply:
Ok.